ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala

```diff
@@ -99,30 +99,31 @@ private[appendonlydao] final class TransactionsReader(
 
     val requestedRangeF = getEventSeqIdRange(startExclusive, endInclusive)
 
-    val query = (range: EventsRange[(Offset, Long)]) => {
-      implicit connection: Connection =>
-        logger.debug(s"getFlatTransactions query($range)")
-        queryNonPruned.executeSqlOrThrow(
-          getTransactions(
-            EventsRange(range.startExclusive._2, range.endInclusive._2),
-            filter,
-            pageSize,
-          )(connection),
-          range.startExclusive._1,
-          pruned =>
-            s"Transactions request from ${range.startExclusive._1.toHexString} to ${range.endInclusive._1.toHexString} precedes pruned offset ${pruned.toHexString}",
-        )
-    }
+    val query = (fullRangeForPruningCheck: EventsRange[Offset]) =>
```

Comment:
In the context where this `query` is passed, the fullRange is static, isn't it? Why do you need to have it as an input to this lambda?